src/v/cluster/partition.cc

```diff
@@ -742,7 +742,7 @@ ss::future<> partition::update_configuration(topic_properties properties) {
     if (
       old_ntp_config.is_archival_enabled() != new_archival
       || old_ntp_config.is_read_replica_mode_enabled()
-           != new_ntp_config.read_replica
+           != new_ntp_config.read_replica.value_or(false)
```

Comment:
it should be impossible to enable or disable read-replica mode, the normal topic can't be converted to read-replica and vice versa

Reply:
Should I remove this check then instead of fixing it up? As I wrote in the commit message (https://github.com/redpanda-data/redpanda/pull/18284/commits/aff708585e723b25da1725ce5d93bab11ad9f128), it was causing archival restarts on unrelated changes when the config was set to `read_replica = std::optional(false)`.

Reply:
Fixed